### PR TITLE
Feature/process request

### PR DESCRIPTION
**Pull request by foji9315**
This PR includes next features
- Process a request from AWSGateway through json request
- Convert json request to xml format
-  Store this xml file into S3 bucket \
- Return URL of S3 object resource

[XML Result of process](https://skt-task.s3.us-east-2.amazonaws.com/reservation-12345.xml)

### CODE REVIEW

File: converter/xmlConverter.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
+import xml.etree.ElementTree as ET
+from xml.dom import minidom
+import logging
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+def convertRequestToXML(reservationRequest):
```
Comment (eso31) [inline converter/xmlConverter.py:9]: Try to have a lowercase snake naming convention for your functions, variables, and filenames.

Comment (foji9315) [reply converter/xmlConverter.py]: Done, I changed all variables and methods to follow snake naming convention.

File: model/reservation.py [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        if confirmation_numbers is None:
+            confirmation_numbers = []
+
+        self.hotel = hotel
+        self.reservationId = reservation_id
+        self.confirmationNumbers = confirmation_numbers
+        self.lastUpdateTimestamp = last_update_timestamp
+        self.lastUpdateOperatorId = last_update_operator_id
+
+    def append_confirmation_number(self, confirmation_number):
```
Comment (eso31) [inline model/reservation.py:17]: is this method being used?

File: lambda_function.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+from converter.xml_converter import convert_request_to_xml
+from repository.xml_repository import upload_xml
+from transformer.request_transformer import transform_event
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+def lambda_handler(event, context):
+    logger.info(context)
```
Comment (eso31) [inline lambda_function.py:13]: Let's say the body is not present or empty and a KeyError/JSONDecodeError is raised, what response is the lambda going to return? should it catch unexpected errors and handle them?

File: repository/xml_repository.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+import logging
+from botocore.exceptions import ClientError
+
+logger = logging.getLogger()
+logger.setLevel(logging.INFO)
+
+
+def upload_xml(xml_text, bucket_name, object_name=None):
+    if object_name is None:
+        object_name = "reservation.xml"
```
Comment (eso31) [inline repository/xml_repository.py:11]: If for some reason this method is called without an object name several times, will `reservation.xml` get overridden every time on s3?

File: model/reservation.py (hunk 2 above)
Comment (foji9315) [reply model/reservation.py:17]: no, it is not used. I am goin to remove it

File: lambda_function.py (hunk 3 above)
Comment (foji9315) [reply lambda_function.py:13]: Error handling, validations, refactoring  and refactoring code is being planned to next feature. for now I am just considering happy paths for any request.

File: repository/xml_repository.py (hunk 4 above)
Comment (foji9315) [reply repository/xml_repository.py:11]: Yes, for now I leaving this as a temporal approach, we can have a meeting in order to clarify which fields in the request are required. I that way we can plan how to name xml files if there are not specific data to name it.

File: model/reservation.py (hunk 2 above)
Comment (foji9315) [reply model/reservation.py]: Done, I removed it

**Review by eso31 — APPROVED** [review]

File: lambda_function.py [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    logger.info('Trying to convert object to XML file')
+
+    xml_text = convert_request_to_xml(reservation_request)
+    logger.info('Result of conversion %s', xml_text)
+
+    is_completed, text = upload_xml(xml_text,
+                                    "skt-task",
+                                    'reservation-{}.xml'.format(reservation_request.reservation.reservationId)
+                                    )
+    if is_completed:
```
Comment (motenrique) [inline lambda_function.py:32]: I see you are returning two values on your function here. What do you think about catching an exception that could be thrown if the upload method fails for some reason? In that way, you avoid to use if/else statements for handling that kind of errors

File: repository/xml_repository.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+    try:
+        client = boto3.client('s3')
+        logger.info("Client meta %s", client.meta)
+
+        response = client.put_object(Body=xml_text, Bucket=bucket_name, Key=object_name, ACL='public-read')
+        logger.info("response from bucket %s", response)
+
+        return True, 'https://{}.s3.{}.amazonaws.com/{}'.format(bucket_name, "us-east-2", object_name)
+
+    except ClientError as e:
```
Comment (motenrique) [inline repository/xml_repository.py:21]: As I mentioned on another comment. You can handle your custom exceptions for indicate something was wrong

File: lambda_function.py (hunk 5 above)
Comment (foji9315) [reply lambda_function.py:32]: I agree, for that purpose I will include error handling in next package of features. I will take in consideration this observation on that  enhancements.

File: repository/xml_repository.py (hunk 6 above)
Comment (foji9315) [reply repository/xml_repository.py:21]: I will do on main flow.

File: lambda_function.py (hunk 5 above)
Comment (foji9315) [reply lambda_function.py]: I already added try except block

**Review by motenrique — APPROVED** [review]

**Pull request merged**